Docs/FHS Bundles.md

```diff
@@ -0,0 +1,123 @@
+# Installed Bundles on UNIX-like Systems
+
+This document covers {CF,NS,}Bundle behavior on Linux, with an eye to keeping things working for non-Linux OSes that have similar POSIX-y setups (in particular, the BSD family.) It covers _two_ different styles of bundle, which are suitable both for things that are installed systemwide (e.g. as part of a platform-supplied package manager), and for freestanding bundles that are embedded (e.g. resource bundles) or executed (e.g. app bundles) but not installed systemwide.
+
+The aim of this proposal is to provide idiomatic ways for POSIX-y systems to use bundle API, including resource access, in a way equivalent to what Darwin OSes do for bundles installed in `/System` or `/Library`.
+
+## Installed bundles
+
+An installed bundle is intended to be installed in a directory hierarchy that models the use of `--prefix=…` in autoconf-like configuration tools. This is suitable for the kind of bundles that we would install in `/System/Library` or in `/Applications`, systemwide, on Darwin OSes: system frameworks, apps installed for all users, and so on.
+
+This setup complies with the [Filesystem Hierarchy Standard (FHS) 3.0](https://refspecs.linuxfoundation.org/fhs.shtml), used by most Linux distributions. It also fits with the intent of the `/usr` and `/usr/local` directories of many BSD systems (e.g. from FreeBSD's [hier(7)][].)
+
+[hier(7)]: https://www.freebsd.org/cgi/man.cgi?hier(7)
+
+### Definition
+
+An installed bundle exists if there is:
+
+ - A directory with the `.resources` extension;
+ - contained in a directory named `share`.
+
+The base name of this bundle is the name of this directory, removing the extension. (E.g., for `/usr/share/MyFramework.resources`, the base name is `MyFramework`.)
+
+### Bundle Paths
+
+The bundle's `.resources` directory contains resources much like an iOS-style flat bundle. Unlike on iOS, however, executables are not contained directly inside this directory. Instead, given that the bundle exists in `share/MyFramework.resources`, then an installed bundle will source from the following additional paths within the prefix the bundle is installed in (the parent directory of the `share` directory):
+
+ - The main executable will be searched in the `bin`, `sbin`, or `lib…` directories.
+	 - Executable bundles will search `bin`, then `sbin`, for an executable with the same base name as the resources folder, plus any platform prefix or suffix for executables. e.g.: `bin/MyApp`.
+	 - Framework bundles will search the appropriate `lib…` directories for a library of the same name, adding appropriate platform prefixes or suffixes for shared library file (falling back to just `lib` otherwise). e.g.: `lib/libMyFramework.so`.
+ - Auxiliary executables and libraries will be in, or in subdirectories of, `libexec/MyFramework.executables`;
+ - For framework bundles, `include/MyFramework` will contain headers, though this directory does affect the return values of the runtime API.
```

Comment:
Will we also support a Modules directory like we do for frameworks? One reason this distinction is important is that a framework module map is written differently (`framework module`) than one for flat headers, and there may be something we want to do to reconcile this (presumably Clang itself isn't going to start understanding this convention).

Reply:
Good question. I'll circle back to you on that.

Reply:
So.

This document _does_ affect the discussion you mention, but right now we're a little more interested in the runtime aspect of this. The compile-time aspect is additive, and we can set that up with the clang team as a follow-up from implementing this — there's no runtime API that is affected by the presence (or absence) of the headers or of the module file, as far as I can tell.

Reply:
That makes sense. I don't see any issues with tackling that later.